src/llmescache/langchain.py

```diff
@@ -0,0 +1,90 @@
+import hashlib
+from datetime import datetime
+from typing import Any, Optional
+
+import elasticsearch
+from langchain_core.caches import RETURN_VAL_TYPE, BaseCache
+from langchain_core.load import dumps
+
+
+class ElasticSearchCache(BaseCache):
+    """Cache store for LLM using ElasticSearch."""
+
+    def __init__(
+        self,
+        es_client: elasticsearch.Elasticsearch,
+        es_index: str,
+        store_input: bool = True,
+        store_datetime: bool = True,
+        additional_metadata: Optional[dict] = None,
+    ):
+        """
+        Initialize the ElasticSearch cache store.
+
+        Parameters:
+            es_client: Elasticsearch
+                The Elasticsearch client to use for the cache store.
+            es_index: str
+                The name of the index to use for the cache store.
+            store_input: bool
+                Whether to store the input in the cache, i.e. the LLM input messages.
+            store_datetime: bool
+                Whether to store the datetime in the cache, i.e. when the first time the input was used.
```

Comment:
"i.e. the time of the first request for an input"

Reply:
changed